src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,39 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    public Figure getRandomFigure() {
+        Random random = new Random();
+        ColorSupplier cs = new ColorSupplier();
+        int index = random.nextInt(4);
+        Figure figure;
+        switch (index) {
+            case 0:
+                figure = new Circle(random.nextInt(100), cs.getRandomColor());
+                break;
+            case 1:
+                figure = new Square(random.nextInt(100), cs.getRandomColor());
+                break;
+            case 2:
+                figure = new Rectangle(random.nextInt(100),
+                        random.nextInt(100), cs.getRandomColor());
+                break;
+            case 3:
+                figure = new IsoscelesTrapezoid(random.nextInt(100),
+                        random.nextInt(100), random.nextInt(100), cs.getRandomColor());
+                break;
+            case 4:
+                figure = new RightTriangle(random.nextInt(100),
+                        random.nextInt(100), cs.getRandomColor());
+                break;
+            default:
```

Comment:
Lets create 5 private methods getRandomCircle(), getRandomSquare(), ... and call them in switch-case construction, this will increase code readability. (Remember private methods should follow after public as helper ones)